user-interface/src/lib/models/api.ts

```diff
@@ -24,16 +47,30 @@ export default class Api {
     return path;
   }
 
+  private static executeBeforeHooks() {
+    for (const hook of beforeHooks) {
+      hook();
+    }
+  }
+
+  private static async executeAfterHooks(response: Response) {
+    for (const hook of afterHooks) {
+      await hook(response);
+    }
+  }
+
   public static async post(
     path: string,
     body: object,
     options?: ObjectKeyVal,
   ): Promise<ResponseData> {
     try {
+      this.executeBeforeHooks();
       const apiOptions = this.getQueryStringsToPassthrough(window.location.search, options);
       const pathStr = Api.createPath(path, apiOptions);
 
       const response = await httpPost({ url: Api.host + pathStr, body, headers: this.headers });
```

Comment:
And to my point earlier... the before hooks aren't really "before" if they are allowed to proceed in the background while moving on to the Post call.